tidy3d/components/simulation.py

```diff
@@ -801,7 +830,6 @@ def validate_pre_upload(self) -> None:
         self._validate_monitor_size()
         self._validate_datasets_not_none()
         self._validate_tfsf_structure_intersections()
```

Comment:
Why not also move the tfsf structure intersections to post init?

Reply:
you just need to move [this line](https://github.com/flexcompute/tidy3d/blob/be4220062c227bfa2b90eb54acdafc9fd11e55c8/tests/test_components/test_simulation.py#L1254) above the simulation init (and you should in general be able to remove the `validate_pre_upload` calls in the tfsf tests.)

Reply:
oh yea, I didn't realize at first that it was this simple. ok I made the change.